source/main.c

```diff
@@ -56,6 +59,86 @@ void dol_alloc(int size)
     }
 }
 
+void load_parse_cli()
+{
+    int path_length = strlen(path);
+    path[path_length - 3] = 'c';
+    path[path_length - 2] = 'l';
+    path[path_length - 1] = 'i';
+
+    kprintf("Reading %s\n", path);
+    FIL file;
+    FRESULT result = f_open(&file, path, FA_READ);
+    if (result != FR_OK)
+    {
+        if (result == FR_NO_FILE)
+        {
+            kprintf("CLI file not found\n");
+        }
+        else
+        {
+            kprintf("Failed to open CLI file\n");
+        }
+        return;
+    }
+
+    size_t size = f_size(&file);
+    kprintf("CLI file size is %iB\n", size);
+
+    if (size <= 0)
+    {
+        kprintf("Empty CLI file\n");
+        return;
+    }
+
+    char *cli = (char *) malloc(size + 1);
+
+    if (!cli)
+    {
+        kprintf("Couldn't allocate memory for CLI file\n");
+        return;
+    }
+
+    UINT _;
+    f_read(&file, cli, size, &_);
+    f_close(&file);
+
+    if (cli[size - 1] != '\0')
+    {
+      cli[size] = '\0';
+      size++;
+    }
+
+    // Parse CLI file
+    // https://github.com/emukidid/swiss-gc/blob/a0fa06d81360ad6d173acd42e4dd5495e268de42/cube/swiss/source/swiss.c#L1236
+    dol_argv[dol_argc] = path;
+    dol_argc++;
+
+    // First argument is at the beginning of the file
+    if (cli[0] != '\r' && cli[0] != '\n')
+    {
+        dol_argv[dol_argc] = cli;
+        dol_argc++;
+    }
+
+    // Search for the others after each newline
+    for (int i = 0; i < size; i++)
+    {
+        if (cli[i] == '\r' || cli[i] == '\n')
+        {
+            cli[i] = '\0';
+        }
+        else if (cli[i - 1] == '\0')
+        {
+            dol_argv[dol_argc] = cli + i;
+            dol_argc++;
+            if (dol_argc >= 1024) break;
```

Comment:
I realize that this is probably just a straight copy-paste from Swiss, but we should print a warning here.

Reply:
Agreed. Good catch.